src/devTools.js

```diff
@@ -112,13 +116,18 @@ export function connect(options = {}) {
     },
     send: (action, payload) => {
       if (action) {
-        send(action, payload, options, 'ACTION', id);
+        send(action, payload, options, 'ACTION', id, hostnamePromise);
       } else {
-        send(undefined, payload, options, 'STATE', id);
+        send(undefined, payload, options, 'STATE', id, hostnamePromise);
       }
     },
     error: (payload) => {
-      socket.emit({ type: 'ERROR', payload, id: socket.id, instanceId: id });
+      // like in the `send` function
+      setTimeout(async () => {
+        // makes sure the connection is established
```

Comment:
Nit: I don't think this simple comments are helpful. It should be clear from the code what's going on, and if it's not, then the identifiers should be renamed instead of adding a simple comment. I think comments make the most sense when they're long and explanatory... then there's no way to explain with just identifier naming

Reply:
I think it's a matter of taste but I don't mind removing this comment, I removed it in my newest changes.